### PR TITLE
Chimera/init commands

### PR DESCRIPTION
**Pull request by laxmorek**
### Short description 📝

Working on #1674

Adds:
- command `tuist dependencies fetch`
- command `tuist dependencies update`
- stubbed `DependenciesController`
- new commands call appropriate methods from `DependenciesController `
- unit tests

### CODE REVIEW

File: Sources/TuistKit/Commands/Dependencies/DependenciesFetchCommand.swift [hunk 1]
```diff
@@ -0,0 +1,22 @@
+import ArgumentParser
+import Foundation
+import TSCBasic
+
+/// A coomand to fetch project's dependencies.
+struct DependenciesFetchCommand: ParsableCommand {
+    static var configuration: CommandConfiguration {
+        CommandConfiguration(commandName: "fetch",
+                             abstract: "Fetches project's dependecies ")
```
Comment (facumenzella) [inline Sources/TuistKit/Commands/Dependencies/DependenciesFetchCommand.swift:9]:
```suggestion
                             abstract: "Fetches project's dependencies defined in Dependencies.swift ")
```

File: Sources/TuistKit/Commands/Dependencies/DependenciesUpdateCommand.swift [hunk 2]
```diff
@@ -0,0 +1,22 @@
+import ArgumentParser
+import Foundation
+import TSCBasic
+
+/// A command to update project's dependencies.
+struct DependenciesUpdateCommand: ParsableCommand {
+    static var configuration: CommandConfiguration {
+        CommandConfiguration(commandName: "update",
+                             abstract: "Updates project's dependencies.")
```
Comment (facumenzella) [inline Sources/TuistKit/Commands/Dependencies/DependenciesUpdateCommand.swift:9]:
```suggestion
                             abstract: "Updates project's dependencies defined in Dependencies.swift.")
```

File: Sources/TuistDependencies/DependenciesController.swift [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+        }
+    }
+}
+
+// MARK: - Dependencies Controlling
+
+/// `DependenciesControlling` controls:
+///     1. Fetching/updating dependencies defined in `./Tuist/Dependencies.swift` by running appropriate dependencies managers (`Cocoapods`, `Carthage`, `SPM`).
+///     2. Compiling fetched/updated depedencies into `.framework.`/`.xcframework.`.
+///     3. Saving complited frameworks uder `./Tuist/Dependencies/*`.
```
Comment (facumenzella) [inline Sources/TuistDependencies/DependenciesController.swift:31]:
```suggestion
///     3. Saving compiled frameworks uder `./Tuist/Dependencies/*`.
```

**Review by facumenzella — APPROVED** [review]
Great work!! 💪

File: Sources/TuistDependencies/DependenciesController.swift [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+        case .unimplemented:
+            return .abort
+        }
+    }
+
+    /// Description.
+    public var description: String {
+        switch self {
+        case .unimplemented:
+            return "Pssst! You have found secret and hidden part of project where we are trying to create something new and unexpected. Chimera!"
```
Comment (pepibumur) [inline Sources/TuistDependencies/DependenciesController.swift:21]: Nitpick: I'd rephrase this to:
```
A standard approach for managing third-party dependencies is being worked on and it'll be available soon.
```

File: Sources/TuistKit/Commands/Dependencies/DependenciesFetchCommand.swift (hunk 1 above)
Comment (pepibumur) [reply Sources/TuistKit/Commands/Dependencies/DependenciesFetchCommand.swift:9]: And I'd say: `Fetches the project's....`

File: Sources/TuistKit/Commands/Dependencies/DependenciesUpdateCommand.swift (hunk 2 above)
Comment (pepibumur) [reply Sources/TuistKit/Commands/Dependencies/DependenciesUpdateCommand.swift:9]: Similar comment here in regards to addint the article to `project`.

File: Sources/TuistDependencies/DependenciesController.swift (hunk 4 above)
Comment (laxmorek) [reply Sources/TuistDependencies/DependenciesController.swift:21]: I wanted to be funny here :P Im replacing with your suggestion :)

**Review by pepibumur — APPROVED** [review]
